ql/src/java/org/apache/hadoop/hive/ql/exec/vector/mapjoin/VectorMapJoinAntiJoinLongOperator.java

```diff
@@ -0,0 +1,315 @@
+
+          // Single-Column Long get key.
+          long currentKey;
+          boolean isNull;
+          if (!joinColVector.noNulls && joinColVector.isNull[batchIndex]) {
+            currentKey = 0;
+            isNull = true;
+          } else {
+            currentKey = vector[batchIndex];
+            isNull = false;
+          }
+
+          // Equal key series checking.
+          if (isNull || !haveSaveKey || currentKey != saveKey) {
+            // New key.
+            if (haveSaveKey) {
+              // Move on with our counts.
+              switch (saveJoinResult) {
+              case MATCH:
+                // We have extracted the existence from the hash set result, so we don't keep it.
+                break;
+              case SPILL:
+                // We keep the hash set result for its spill information.
+                hashSetResultCount++;
+                break;
+              case NOMATCH:
+                break;
+              }
+            }
+
+            if (isNull) {
+              saveJoinResult = JoinUtil.JoinResult.MATCH;
+              haveSaveKey = false;
+            } else {
+              // Regardless of our matching result, we keep that information to make multiple use
+              // of it for a possible series of equal keys.
+              haveSaveKey = true;
+              saveKey = currentKey;
+              if (useMinMax && (currentKey < min || currentKey > max)) {
+                // Key out of range for whole hash table, is a valid match for anti join.
+                saveJoinResult = JoinUtil.JoinResult.NOMATCH;
+              } else {
+                saveJoinResult = hashSet.contains(currentKey, hashSetResults[hashSetResultCount]);
+              }
+
+              // Reverse the match result for anti join.
+              if (saveJoinResult == JoinUtil.JoinResult.NOMATCH) {
+                saveJoinResult = JoinUtil.JoinResult.MATCH;
+              } else if (saveJoinResult == JoinUtil.JoinResult.MATCH) {
+                saveJoinResult = JoinUtil.JoinResult.NOMATCH;
+              }
+            }
+
+            // Common anti join result processing.
+            switch (saveJoinResult) {
+            case MATCH:
+              allMatchs[allMatchCount++] = batchIndex;
+              // VectorizedBatchUtil.debugDisplayOneRow(batch, batchIndex, CLASS_NAME + " MATCH isSingleValue " + equalKeySeriesIsSingleValue[equalKeySeriesCount] + " currentKey " + currentKey);
+              break;
+
+            case SPILL:
+              spills[spillCount] = batchIndex;
+              spillHashMapResultIndices[spillCount] = hashSetResultCount;
+              spillCount++;
+              break;
+
+            case NOMATCH:
+              // VectorizedBatchUtil.debugDisplayOneRow(batch, batchIndex, CLASS_NAME + " NOMATCH" + " currentKey " + currentKey);
+              break;
+            }
+          } else {
+            // Series of equal keys.
```

Comment:
Repeating Common anti join result processing code -- move to function?

Reply:
done